### PR TITLE
Skip test when atomic operations are not supported on GPU.

### PR DESCRIPTION
**Pull request by drivanov**
## Description

Following tests 
```
FAILED tests/python/common/ops/test_ops.py::test_gather_mm_idx_b[dtype0-0.01-256] - RuntimeError: CUDA error: unspecified launch failure
FAILED tests/python/common/ops/test_ops.py::test_gather_mm_idx_b[dtype0-0.01-64] - RuntimeError: CUDA error: unspecified launch failure
FAILED tests/python/common/ops/test_ops.py::test_gather_mm_idx_b[dtype0-0.01-16] - RuntimeError: CUDA error: unspecified launch failure
FAILED tests/python/common/ops/test_ops.py::test_gather_mm_idx_b[dtype0-0.01-8] - RuntimeError: CUDA error: unspecified launch failure
FAILED tests/python/common/ops/test_ops.py::test_gather_mm_idx_b[dtype0-0.01-1] - RuntimeError: CUDA error: unspecified launch failure
```
are failing for `dtype=torch.bfloat16` with the error message:
```
Atomic operations are not supported for bfloat16 (BF16) on GPUs with compute capability less than 8.0.
```

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by chang-l** [comment]
Btw, let me provide a bit more background here: 

Recently, PyTorch introduces this commit: https://github.com/pytorch/pytorch/commit/fc5fda14bcc41008e5610bf39c53eb066933ea4e, which changes the behavior of `torch.cuda.is_bf16_supported()`. Previously, `torch.cuda.is_bf16_supported()` will be `False` for any GPUs older than A100 (<sm80). But after this commit, it could return `True` even with V100 GPUs (sm70).  As a result, it can bypass the check here even with a V100 (sm70) GPU: https://github.com/dmlc/dgl/blob/8e6cbd621f34c0892ab5adac5259cd61401a6d2b/tests/python/common/ops/test_ops.py#L410-L415
Sadly, even if we know that A100 is the first gen GPU that supports bf16 operations [[1]](https://en.wikipedia.org/wiki/Ampere_(microarchitecture)) [[2]](https://images.nvidia.com/aem-dam/en-zz/Solutions/data-center/nvidia-ampere-architecture-whitepaper.pdf), we could not use this API(`torch.cuda.is_bf16_supported()`) to query for the bf16 operation support any more.  cc. @TristonC @nv-dlasalle @frozenbugs 

As of now, this commit has not been populated to any pytorch release branch yet. So, it can only be reproduced from a nightly PyTorch build.

**Comment by nv-dlasalle** [comment]
@chang-l to clarify the issue, because DGL has some operations only implemented via atomics, we cannot support these GPUs, but pytorch has non-atomic versions of its operators and thus can implement operations for BF16 on these GPUs?

**Comment by chang-l** [comment]
@nv-dlasalle 
After checking `cuda_bf16.h` (12.3) and the [doc](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH____BFLOAT162__ARITHMETIC.html#group__CUDA__MATH____BFLOAT162__ARITHMETIC_1g550f52c89d672213390e9bfd8a3c42bf), it seems starting from cuda12.2, an 'emulated' support of many bf16 operations have been added to older devices (sm70), which includes atomicAdd operation. I think, technically, all sm70-sm80 devices should support bf16 operations via 'emulation path' from driver side (with cuda 12.2+). 

Then, do we need to update https://github.com/dmlc/dgl/blob/8e6cbd621f34c0892ab5adac5259cd61401a6d2b/src/array/cuda/atomic.cuh to remove the assertions?  cc. @yaox12

**Comment by frozenbugs** [comment]
@dgl-bot

**Comment by Rhett-Ying** [comment]
@dgl-bot

File: tests/python/common/ops/test_ops.py [hunk 1]
```diff
@@ -413,14 +413,14 @@ def test_gather_mm_idx_b(feat_size, dtype, tol):
...
         pytest.skip("BF16 is not supported.")
-
-    if (
-        F._default_context_str == "gpu"
-        and dtype == torch.float16
+    if F._default_context_str == "gpu" and (
+        dtype == torch.float16
         and torch.cuda.get_device_capability() < (7, 0)
+        or dtype == torch.bfloat16
+        and torch.cuda.get_device_capability() < (8, 0)
```
Comment (frozenbugs) [inline tests/python/common/ops/test_ops.py:420]: Sorry I am not good at understanding the priority of `or` and `and`, can you add parentheses?

**Comment by chang-l** [comment]
I am okay with the current fix in this PR.

**Review by chang-l — APPROVED** [review]

**Comment by chang-l** [comment]
@dgl-bot

File: tests/python/common/ops/test_ops.py (hunk 1 above)
Comment (drivanov) [reply tests/python/common/ops/test_ops.py]: Done

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**